core/domain/learner_progress_services_test.py

```diff
@@ -522,6 +597,24 @@ def test_get_all_incomplete_exp_ids(self):
             learner_progress_services.get_all_incomplete_exp_ids(
                 self.user_id), [self.EXP_ID_0, self.EXP_ID_1])
 
+        # Mark EXP_ID_2 as completed and create summary list to test output.
+        learner_progress_services.mark_exploration_as_completed(
+            self.user_id, self.EXP_ID_2)
+        exploration_ids = [self.EXP_ID_0, self.EXP_ID_1, self.EXP_ID_2]
+        exploration_summaries = exp_services.get_exploration_summaries_matching_ids( # pylint: disable=line-too-long
+            exploration_ids)
+
+        # Get all filtered summaries.
+        all_filtered_summaries = self._test_add_private_exploration_activity()
+        incomplete_exp_summaries = all_filtered_summaries.incomplete_exp_summaries # pylint: disable=line-too-long
+
+        # Test that completed exp summaries don't include private exploration.
+        self.assertEqual(
+            incomplete_exp_summaries[0].id, exploration_summaries[0].id)
```

Comment:
I'm not sure why we are testing exploration summaries here (in the learner_progress_services_test.py)? Can you please explain?

Reply:
Of course! 

We start with all the exploration_summaries of EXP_ID_0, EXP_ID_1, and EXP_ID_2. (In this block of code, they all happen to be incomplete, though this is not important). Then we pass these 3 exploration_summaries to _test_add_private_exploration_activity() which makes EXP_ID_2 private. Therefore, when we call get_activity_progress in _test_add_private_exploration_activity(), we are testing to ensure that EXP_ID_2 is not in the final list of summaries (incomplete_exp_summaries) returned by get_activity_progress (since it is private). It should, instead, contain ONLY EXP_ID_0 and EXP_ID_1.

In the 3 assert statements, we first ensure that EXP_ID_0 is in the list. Then we ensure that EXP_ID_1 is in the list. Then we ensure that the list only contains 2 items (aka EXP_ID_2 is not in it).